### PR TITLE
negative sign matching in mbe matching for literal

### LINKED ISSUE #6292: Macro error when using negative literals
I have a macro like this:
```
macro_rules! point {
    ($x:literal, $y:literal, $z:literal) => {
        Vec3::new($x, $y, $z)
    };
}
```
It works fine if I use `point!(0.0, 1.0, 0.0)` but `rust-analyzer` raises an error when the parameter is negative, e.g., `point!(0.0, -1.0, 0.0)`.

Error message:
> Expected 3 arguments, found 1

The same code works in `rustc` without any error or warning.

### PR DESCRIPTION
**Pull request by edwin0cheng**
Fix #6292

r @jonas-schievink

### CODE REVIEW

**Comment by matklad** [comment]
:heart_eyes: welcome back @edwin0cheng  :-)

File: crates/mbe/src/mbe_expander/matcher.rs [hunk 1]
```diff
@@ -356,6 +356,18 @@ impl<'a> TtIter<'a> {
             ExpandResult { value: _, err: Some(_) } => None,
         }
     }
+
+    pub(crate) fn eat_char(&mut self) -> Option<tt::TokenTree> {
```
Comment (jonas-schievink) [inline crates/mbe/src/mbe_expander/matcher.rs:360]: This should probably be named `eat_minus`, or take a `char` parameter that tells it which character to expect

**Review by jonas-schievink — APPROVED** [review]
Looks good, minus a small nit

File: crates/mbe/src/mbe_expander/matcher.rs (hunk 1 above)
Comment (edwin0cheng) [reply crates/mbe/src/mbe_expander/matcher.rs]: Ah... fixed.

**Pull request merged**